### ISSUE TITLE
How to use the library in a create-react-app project?

### ISSUE THREAD

**Issue by rosenk** [aws/amazon-cognito-identity-js#128]
Hi,

My project is seeded with create-react-app tool which is not exposing any webpack.config.js so I cant apply the README instructions directly.

I'm trying to workaround it with:
`import * as AWS from 'exports?AWS!aws-sdk/dist/aws-sdk'`
which causes a warning: 

```
Critical dependencies:                                                                                                                                                          
4:113-120 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.                
 @ ./~/exports-loader?AWS!./~/aws-sdk/dist/aws-sdk.js 4:113-120 
```

and
`import {AuthenticationDetails, CognitoUser} from 'amazon-cognito-identity-js';`
which causes an error:

```
Error in ./~/aws-sdk/lib/api_loader.js                                                                                                                                          
Module not found: Error: Cannot resolve module 'fs' in /home/rosen/home/dev/meoptions/webportal/node_modules/aws-sdk/lib                                                        
 @ ./~/aws-sdk/lib/api_loader.js 1:9-22                                                                                                                                         

Error in ./~/aws-sdk/lib/services.js                                                                                                                                            
Module not found: Error: Cannot resolve module 'fs' in /home/rosen/home/dev/meoptions/webportal/node_modules/aws-sdk/lib                                                        
 @ ./~/aws-sdk/lib/services.js 1:9-22   
```

Please help.

**Comment by crediblenotand** [comment]
`npm run eject` will deconstruct the react-scripts into its individual parts, including the webpack config, at which point you can follow the instructions successfully.

Maybe there's a way to merge/enhance the webpack config without having to "eject" the react-scripts, which would be ideal (although somewhat flakey, in the scenario where react-scripts replaces webpack with rollup)

**Comment by rosenk** [comment]
I'm thinking about encapsulating all my Cognito related code in another small project which will have a webpack.config.js of it's own. Then I'll add somehow the dependency to it in my primary package.json. I don't know is it going to work. Hopefully there is a better solution.

**Comment by rosenk** [comment]
Maybe amazon-cognito-identity-js is such external project that should encapsulate some of the problems. Is it possible to apply the aws-sdk workarounds in it own webpack.config.js than instruct users in modify theirs?

**Comment by crediblenotand** [comment]
> Hopefully there is a better solution.
- there's also just including the dependencies from the html file: https://github.com/aws/amazon-cognito-identity-js#install-using-seperate-javascript-files but I had problems with that too
- we could also add a new build script in this project to generate a new `CognitoIdentityServiceProvider` that didn't require a webpack configuration

**Comment by simonbuchan** [comment]
The underlying issue is that the default `aws-sdk` browser main (`lib/browser.js`) is intended for browserify, which seems to treat `require(someExpr)` differently. See aws/aws-sdk-js#603

I'm currently using a weirder setup: https://github.com/aws/aws-sdk-js/issues/603#issuecomment-240905199 but that issue has a bunch of configs you can try.

@rosenk Unfortunately, AFAIK, packages can't change how webpack treats other packages, the only thing it can do to itself is have a `webpack` entry in it's package.json to use as a different main if needed.

@crediblenotand:
- Script usage should still work - I broke them with #108 but fixed them it #112 - maybe open an issue for what is confusing or broken?
- `/dist/aws-cognito-sdk.js` is built from the standard `aws-sdk` (with it's `dist-tools`), so you would essentially be fixing aws/aws-sdk-js#603

**Comment by hengbenkeji** [comment]
Refer to #123 on the setup for React and Webpack.
